test/unittests/state_transition.cpp

```diff
@@ -17,8 +17,26 @@ void state_transition::SetUp()
 
 void state_transition::TearDown()
 {
-    auto& state = pre;
+    auto state = pre;
     const auto res = evmone::state::transition(state, block, tx, rev, vm, block.gas_limit);
+
+    if (expect.tx_error != SUCCESS)
+    {
+        ASSERT_TRUE(holds_alternative<std::error_code>(res));
```

Comment:
Why is this assertion and not proper check, what if implementation mistakenly does successful transition for invalid transaction?

Reply:
There is another old check after this block which checks the other part. But probably make sense to rework this.

Reply:
Hm... I'm not sure how to do this better. Any suggestions?

Reply:
Actually I confused this with normal assertion. It think this is fine then, maybe just add a better message
```cpp
ASSERT_TRUE(holds_alternative<std::error_code>(res))) << "tx expected to be invalid with error: " << expected.message();
```